addons/l10n_my_edi/models/account_move.py

```diff
@@ -0,0 +1,860 @@
+        The format is also the one exported to the platform by the seller, so this could need some tweaks depending on the way it's sent.
+        """
+        # We start by loading the json
+        json_data = json.loads(document_string)
+        invoice_data = json_data['Invoice']
+
+        # The logic is as follows: if an element has sub-elements, it is using the CAC namespace. If not, it is using CBC.
+        # For CAC elements, if multiples are present (e.g. multiple invoice lines) you will file a single InvoiceLine element with multiple child items.
+        root = self._l10n_my_make_sub_element('Invoice', invoice_data[0], root=True)
+        return etree.tostring(root).decode()
+
+    def _l10n_my_make_sub_element(self, element_name, element_data, root=False):
+        """ Add a given node, to the given element.
+        Used recursively to populate the xml tree from the root, based on the given json element
+        """
+        # These are not relevant to the import of invoices, and it would be hard to transform these
+        node_to_ignore = ['UBLExtensions', 'Signature']
+
+        # Start by making the element
+        is_leaf = not any(isinstance(value, list) for value in element_data.values())
+        if not root:
+            if is_leaf:
+                element_name = f"{{{NAMESPACES.get('cbc')}}}{element_name}"
+            else:
+                element_name = f"{{{NAMESPACES.get('cac')}}}{element_name}"
+
+        namespaces = None
+        if root:
+            namespaces = NAMESPACES
+
+        if not is_leaf:
+            # A leaf could have attributes. In this case, they are elements that are not lists.
+            element = etree.Element(element_name, nsmap=namespaces)
+            for sub_element_name, sub_element_data in element_data.items():
+                if sub_element_name in node_to_ignore:
+                    continue
+
+                if isinstance(sub_element_data, list):
+                    for sub_element in sub_element_data:
+                        element.append(self._l10n_my_make_sub_element(sub_element_name, sub_element))
+                else:
+                    # It's actually an attribute
+                    element.set(sub_element_name, sub_element_data)
+        else:
+            # the '_' key hold the text, while 'named' key hold the attribute
+            value = element_data.pop('_', '')
+            # The remaining data are attributes and their value. They can be passed to Element to build the element
+            element = etree.Element(element_name, **element_data, nsmap=namespaces)
+            if not isinstance(value, str):
+                value = str(value)
+            element.text = value
+
+        return element
+
+    def _l10n_my_edi_cancel_moves(self):
+        """ Try to cancel the moves in self if allowed by the lock date. """
+        for move in self:
+            move._check_fiscalyear_lock_date()
+            move.line_ids._check_tax_lock_date()
```

Comment:
Isn't this going to give a weird error to the user?
Say, they update the status, and then get a lock date error.
Maybe this is the lesser of two evils (if locked entries really really must not be changed)
but in my view this is going to be quite confusing for users.

Reply:
It could be.
This is a mistake I made in VN too anyway; we don't want to raise here.
It would cause losing some important information in case we are cancelling using the wizard, it succeeds on the platform but fails in Odoo for any reason.

In practice, these should never fail as there is a 72-hour window for these changes and it is unlikely that you lock a period so close to its end date.

I added a try and catch around the lines; and if the cancellation fails it will post a message instead to warn about it.